src/url.ts

```diff
@@ -118,24 +118,34 @@ export class Url {
     return {...(this.parts.params || {})};
   }
 
-  get nonNullParams(): Record<string, string> {
-    return Object.keys(this.parts.params || {}).reduce<Record<string, string>>((memo, key) => {
-      if (this.parts.params![key] != null) {
-        memo[key] = this.parts.params![key]!.toString();
-      }
+  get nonNullParams(): Record<string, string[]> {
+    const params = this.parts.params || {};
+
+    return Object.keys(params).reduce<Record<string, string[]>>(
+      (memo, key) => {
+        const value = params[key];
+
+        if (value != null) {
+          memo[key] ? memo[key].push(value.toString()) : (memo[key] = [value.toString()]);
+        }
 
-      return memo;
-    }, {});
+        return memo;
+      },
+      {} as Record<string, string[]>
+    );
   }
```

Comment:
Why not just filter params with empty array value ?

```ts
class Url {
  // ...

  get nonNullParams(): TUrlParams {
    const {params = {}} = this.parts;

    return Object.keys(params).reduce((result: TUrlParams , name: string) => {
      const value = params[name];

      if (value && value.length) {
        result[name] = value;
      }

      return result;
    }, {} as TUrlParams);
  }

  // ...
}
```

Reply:
Not sure this is critical functionality for this class to return filtered params, so most probably we can drop this property as we already do a major change in this library

Reply:
it will not work for `?color[]=white&color[]=black` 
we can have multiple params with same name

Reply:
Below It's how I'm expecting it to work: did I miss sth ?

```ts
const regular = Url.fromString("https://example.com?color[]=white&color[]=black");

regular.params // {color: ["white", "black"]}

const withEmptyParam = Url.fromString("https://example.com?color[]=white&color[]=black&color[]=");

withEmptyParam.params // {color: ["white", "black", ""]}
```